### PR TITLE
Feature/integration tests to backend

### PR DESCRIPTION
**Pull request by fipa98**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,23 @@ container/*/sql
 .vscode
 
 frontend/public/bundle.js
+
+backend/.vs/backend/config/applicationhost.config
```
Comment (petterroea) [inline .gitignore]: This is very messy. `backend/.vs` should be gitignored.

File: XUnitTest_backend/ControllerTests/MunicipalityControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+
+namespace OpenData
+{
+    public class MunicipalityControllerTests
+    {
+        // Municipality tests here
```
Comment (petterroea) [inline XUnitTest_backend/ControllerTests/MunicipalityControllerTests.cs]: What is this?

File: XUnitTest_backend/ControllerTests/UserControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Xunit;
+
+
+namespace OpenData
+{
+    public class UserControllerTests
+    {
+        // User tests here
```
Comment (petterroea) [inline XUnitTest_backend/ControllerTests/UserControllerTests.cs]: What is this?

**Comment by petterroea** [comment]
Still need 2 reviews. @Letaguator others?

File: XUnitTest_backend/BasicTests.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly CustomWebApplicationFactory<Startup> _factory;
+
+        public BasicTests(CustomWebApplicationFactory<Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        [Theory]
+        [InlineData("/api/municipality")]
+        public async Task Get_EndpointsReturnSuccessAndCorrectContentType(string url)
```
Comment (Letaguator) [inline XUnitTest_backend/BasicTests.cs]: What is up with non Pascal case?

File: backend/backend.csproj [hunk 5]
```diff
@@ -7,6 +7,7 @@
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="9.0.0" />
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="7.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="3.1.1" />
```
Comment (Letaguator) [inline backend/backend.csproj]: This is not needed in backend?

File: backend/backend.sln [hunk 6]
```diff
@@ -13,5 +16,15 @@ Global
 		{FA31E371-D27D-4A32-A81F-64BDCA653835}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{FA31E371-D27D-4A32-A81F-64BDCA653835}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{FA31E371-D27D-4A32-A81F-64BDCA653835}.Release|Any CPU.Build.0 = Release|Any CPU
+		{2B0A2414-5584-4C50-9FDB-3AB1E7974226}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{2B0A2414-5584-4C50-9FDB-3AB1E7974226}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{2B0A2414-5584-4C50-9FDB-3AB1E7974226}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{2B0A2414-5584-4C50-9FDB-3AB1E7974226}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
```
Comment (Letaguator) [inline backend/backend.sln]: These changes needs to be looked at @petterroea

**Pull request merged**